src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.BeforeCommon.targets

```diff
@@ -137,6 +137,11 @@ Copyright (c) .NET Foundation. All rights reserved.
     <DocumentationFile />
   </PropertyGroup>
 
+  <PropertyGroup Condition="'$(PublishDocumentationFile)' == ''">
+    <PublishDocumentationFile Condition="'$(GenerateDocumentationFile)' == 'true'">true</PublishDocumentationFile>
```

Comment:
I think it should be false by default regardless. Apps can opt-in and packages that depend on it (like swagger) could even carry props to turn it on. It's consistent with the v1 behavior and it's pay-for-play in terms of publish size.